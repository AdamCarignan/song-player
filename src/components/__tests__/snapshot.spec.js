import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import SongItem from '@/components/SongItem.vue'

describe('Snapshot SongItem.vue', () => {
  test('Renders Correctly', () => {
    const song = {
      docId: 'abc',
      modified_name: 'song name modified',
      display_name: 'song name',
      comment_count: 3
    }
    const wrapper = shallowMount(SongItem, {
      shallow: true,
      propsData: {
        song
      },
      global: {
        components: {
          'router-link': RouterLinkStub
        }
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
