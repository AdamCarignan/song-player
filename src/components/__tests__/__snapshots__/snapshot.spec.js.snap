// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot SongItem.vue > Renders Correctly 1`] = `
<div
  id="song-id-abc"
>
  <div>
    <router-link-stub
      class="font-bold block text-gray-600"
      custom="false"
      to="[object Object]"
    />
    <span
      class="text-gray-500 text-sm"
    >
      song name
    </span>
  </div>
  <div
    class="text-gray-600 text-lg"
  >
    <router-link-stub
      custom="true"
      to="[object Object]"
    />
  </div>
</div>
`;
